sunpy/net/dataretriever/sources/goes.py

```diff
@@ -58,18 +58,28 @@ def _get_url_for_timerange(self, timerange, **kwargs):
             Data type to return for the particular GOES satellite. Supported
             types depend on the satellite number specified. (default = xrs_2s)
         """
-
+        start_date = parse_time(timerange.start)
+        end_date = parse_time(timerange.end)
+        
         # find out which satellite and datatype to query from the query times
         sat_num = self._get_goes_sat_num(timerange.start, timerange.end)
         base_url = 'http://umbra.nascom.nasa.gov/goes/fits/'
 
-        if timerange.start < parse_time('1999/01/15'):
-            url = base_url + "{date:%Y}/go{sat:02d}{date:%y%m%d}.fits".format(
-                date=timerange.start, sat=sat_num[0])
-        else:
-            url = base_url + "{date:%Y}/go{sat:02d}{date:%Y%m%d}.fits".format(
-                date=timerange.start, sat=sat_num[0])
-        return [url]
+        total_days = (end_date - start_date).days + 1
```

Comment:
We never decided on a standard for time ranges did we?  Closed on the left,
open on the right?

On Wed, Apr 27, 2016 at 5:01 AM, Stuart Mumford <notifications@github.com>
wrote:

> In sunpy/net/dataretriever/sources/goes.py
> <https://github.com/sunpy/sunpy/pull/1747#discussion_r61224976>:
>
> >          # find out which satellite and datatype to query from the query times
> >          sat_num = self._get_goes_sat_num(timerange.start, timerange.end)
> >          base_url = 'http://umbra.nascom.nasa.gov/goes/fits/'
> >
> > -        if timerange.start < parse_time('1999/01/15'):
> > -            url = base_url + "{date:%Y}/go{sat:02d}{date:%y%m%d}.fits".format(
> > -                date=timerange.start, sat=sat_num[0])
> > -        else:
> > -            url = base_url + "{date:%Y}/go{sat:02d}{date:%Y%m%d}.fits".format(
> > -                date=timerange.start, sat=sat_num[0])
> > -        return [url]
> > +        total_days = (end_date - start_date).days + 1
>
> That's true but this download code should include the end time, it's for
> the logic further up in the chain (in attrs.Time) to deal with excluding
> bits of time ranges.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/sunpy/sunpy/pull/1747/files/eb539c6a5af662f6f35875c49c91982b028c0af5#r61224976>
>